src/series/ops/image.rs

```diff
@@ -1,19 +1,26 @@
-use crate::datatypes::{DataType, ImageType};
+use arrow2;
+
+use crate::datatypes::{DataType, ImageType, NullArray};
 
-use crate::series::IntoSeries;
 use crate::{
     error::{DaftError, DaftResult},
-    series::Series,
+    series::{IntoSeries, Series},
 };
 
 impl Series {
     pub fn image_decode(&self) -> DaftResult<Series> {
         match self.data_type() {
             DataType::Binary => {
-                todo!("not implemented");
+                let binary_array = self.binary()?;
+                if binary_array.data().null_count() == self.len() {
```

Comment:
This may cause some divergences across partitions during runtime. we should just take the same path.

Reply:
I tried keeping a single `ImageArray` path, but we can't create a well-typed `ImageArray` since we can't infer the pixel dtype from the encoded bytes if it's all nulls, so we can't construct a `DataType::Image()` dtype or the underlying struct array.

The only thing that I can think of is creating a `DataType::Image(DataType::Null)` dtype and using a `NullArray` as the underlying data array, and try to make it so this doesn't break any downstream invariants we might have around the pixel dtype/array?

Reply:
@samster25 I kept the top-level array type as an `ImageArray` and added an optimized all-null path at the Arrow level.